engine/consensus/matching/core.go

```diff
@@ -188,7 +187,10 @@ func (c *Core) OnReceipt(originID flow.Identifier, receipt *flow.ExecutionReceip
 		// recursively processing the child receipts, since onReceipt
 		// is logging error internal already, we could ignore the returned
 		// error here
-		_ = c.OnReceipt(childReceipt.ExecutorID, childReceipt)
+		err := c.OnReceipt(childReceipt.ExecutorID, childReceipt)
+		if err != nil {
+			return err
```

Comment:
```suggestion
			// we don't want to wrap the error with any info from its parent receipt, because the error
			// has nothing to do with its parent receipt. 
			return err
```